src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -562,6 +564,21 @@ private void UpdateCallsiteExecutionState(Object callsiteData, RuntimeCore runti
                 }
             }*/
         }
+        //TODO write a unit test for this.
+        private static bool CheckIfTraceDataIsLegacySOAPFormat(string base64EncodedTraceData)
+        {
+            var data = Convert.FromBase64String(base64EncodedTraceData);
+            if (data.Length > 71)
```

Comment:
What are the first 71 chars? Does it include `<?xml version=”“1.0"” encoding=““utf-8"”?> ...`?

Reply:
it's `<SOAP-ENV:Envelope xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"` exactly.

Reply:
to clarify the reason I do this check is because if theres less than 71 bytes, the `GetString()` call below will fail.